install/install.go

```diff
@@ -1172,7 +1173,10 @@ func (k *K8sInstaller) getCiliumVersion() semver.Version {
 		ersion = strings.TrimPrefix(k.params.Version, "v")
 	}
 	v, err := versioncheck.Version(ersion)
-	if err != nil {
+	switch {
+	case k.params.DevMode:
+		return v
+	case err != nil:
```

Comment:
I think this part of the change will break installation (i.e. creation of the ConfigMap wrt. `enable-ipv4-masquerade` vs. `masquerade`) when running development versions build from Cilium `v1.9` branch or earlier, see here:

https://github.com/cilium/cilium-cli/blob/3b177845c18812045d92e6bc398b9d6be6f9d986/install/install.go#L1319-L1324

In this case, I think we should instead rely on the developer specifying the correct base version using the `--base-version` flag and if not specified assume >=1.10.0 even in `--dev` mode.